src/GdbConnection.cc

```diff
@@ -1258,8 +1258,9 @@ bool GdbConnection::process_packet() {
 }
 
 void GdbConnection::notify_no_such_thread(const GdbRequest& req) {
-  DEBUG_ASSERT(!memcmp(&req, &this->req, sizeof(this->req)));
-
+  DEBUG_ASSERT(!memcmp(&req.target, &this->req.target, sizeof(req.target)) &&
```

Comment:
Why not use `operator==` instead of `memcmp`?

Reply:
I didn't realize it had one. This was more of a drive-by commit ;). Fixed.